Crash when attempting to sync the profile with mobile on gentoo

# Bug Report

## Description

Experienced by a user.

Their logs sadly don't show much:
```
INF 2025-03-27 18:50:03.939Z fetch mailservers                          topics="mailservers-service" tid=21290 file=service.nim:200 rpc_proc=mailservers_getMailservers response="(jsonrpc: \"2.0\", result: null, id: 0, error: nil)"
WRN 2025-03-27 18:50:04.069Z missing locale details                     tid=21290 file=module.nim:42 locale=base
WRN 2025-03-27 18:50:04.083Z expected response is not a json object     topics="ens-service" tid=21290 file=service.nim:231 procName=getAllMyEnsUsernames
WARN [03-27|18:50:04.397] Served ens_getName                       reqid=32 duration=350.884784ms err="not a resolver"
ERR 2025-03-27 18:50:04.396Z rpc response error                         topics="rpc" tid=21316 file=core.nim:36 err="\nstatus-go error [methodName:ens_getName, code:-32000, message:not a resolver ]\n"
ERR 2025-03-27 18:50:04.396Z error doing rpc request                    topics="rpc" tid=21316 file=core.nim:39 methodName=ens_getName exception="\nstatus-go error [methodName:ens_getName, code:-32000, message:not a resolver ]\n"
ERR 2025-03-27 18:50:04.396Z error:                                     topics="wallet-account-service" tid=21316 file=utils.nim:75 errDesription="\nstatus-go error [methodName:ens_getName, code:-32000, message:not a resolver ]\n"
WARN [03-27|18:50:04.683] Served ens_getName                       reqid=46 duration=284.476926ms err="not a resolver"
ERR 2025-03-27 18:50:04.679Z rpc response error                         topics="rpc" tid=21316 file=core.nim:36 err="\nstatus-go error [methodName:ens_getName, code:-32000, message:not a resolver ]\n"
ERR 2025-03-27 18:50:04.679Z error doing rpc request                    topics="rpc" tid=21316 file=core.nim:39 methodName=ens_getName exception="\nstatus-go error [methodName:ens_getName, code:-32000, message:not a resolver ]\n"
ERR 2025-03-27 18:50:04.679Z error:                                     topics="wallet-account-service" tid=21316 file=utils.nim:75 errDesription="\nstatus-go error [methodName:ens_getName, code:-32000, message:not a resolver ]\n"
INFO [03-27|18:50:07.547] New local node record                    seq=1,743,101,402,491 id=4c5b3c5b375af949 ip=<nice try> udp=52956 tcp=46219
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
Segmentation fault
```

## Steps to reproduce

> when attempting to sync profile with mobile
> desktop syncing from mobile
> via code and not qr

#### Expected behavior

Works

#### Actual behavior

Crashes

### Additional Information

- Status desktop version: latest (2.33)
- Operating System: gentoo amd64 glibc(!) openrc

I can reproduce when I try to pair my two desktop apps

```
DBG 2025-07-24 17:18:01.766Z local pairing event                        topics="signals-manager" tid=1934605 file=pairing.nim:26 type=EventProcessSuccess action=ActionPairingAccount error= event="{\"type\":\"localPairing\",\"event\":{\"type\":\"process-success\",\"action\":2},\"timestamp\":1753377481}"
DBG 2025-07-24 17:18:02.116Z local pairing event                        topics="signals-manager" tid=1934605 file=pairing.nim:26 type=EventTransferSuccess action=ActionSyncDevice error= event="{\"type\":\"localPairing\",\"event\":{\"type\":\"transfer-success\",\"action\":3},\"timestamp\":1753377482}"
WRN 2025-07-24 17:18:02.957Z Error decoding signal                      topics="signals-manager" tid=1934605 file=signals_manager.nim:49 err="Unknown signal received: mediaserver.started"
INFO [07-24|17:18:02.958] Starting peer-to-peer node               instance=StatusIM/vv10.29.0-140-g9a4ada9a5/linux-amd64/go1.23.11
WARN [07-24|17:18:02.958] P2P server will be useless, neither dialing nor listening 
INFO [07-24|17:18:02.962] New local node record                    seq=1,753,377,482,962 id=1b634a3603e67959 ip=127.0.0.1 udp=0 tcp=0
INFO [07-24|17:18:02.962] Started P2P networking                   self=enode://4767220df9ed8a7e35223fe50bc85905353f537715ea81ae914d06c8cb650bba5e1c28c323f53854e86926da548a34896223842e4a7e22afcdabf3e326a0efc0@127.0.0.1:0
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
make: *** [Makefile:893: run-linux] Segmentation fault (core dumped)
```

--- a/protocol/messenger.go
+++ b/protocol/messenger.go
@@ -970,6 +970,10 @@ func (m *Messenger) handleConnectionChange(online bool) {
 }
 
 func (m *Messenger) Online() bool {
+	if !m.started {
+		return false
+	}
+
 	if m.config.onlineChecker != nil {
 		return m.config.onlineChecker()
 	}

--- a/protocol/messenger_communities.go
+++ b/protocol/messenger_communities.go
@@ -1032,6 +1032,9 @@ func (m *Messenger) JoinCommunity(ctx context.Context, communityID types.HexByte
 }
 
 func (m *Messenger) subscribeToCommunityShard(communityID []byte, shard *messagingtypes.Shard) error {
+	if !m.started {
+		return nil
+	}
 	// TODO: this should probably be moved completely to transport once pubsub topic logic is implemented
 	pubsubTopic := shard.PubsubTopic()
 

--- a/tests-functional/tests/test_local_pairing.py
+++ b/tests-functional/tests/test_local_pairing.py
@@ -184,6 +184,9 @@ class TestLocalPairing(MessengerSteps):
         # Make contacts before local pairing
         self.make_contacts(alice, bob)
 
+        # Create community before local pairing
+        self.create_community(bob)
+
         # Local pairing
         pair_server_as_sender(bob, bob_second_device)
 
@@ -215,6 +218,9 @@ class TestLocalPairing(MessengerSteps):
 
         # Make contacts before local pairing
         self.make_contacts(alice, bob)
+
+        # Create community before local pairing
+        self.create_community(bob)
 
         # Local pairing
         pair_server_as_receiver(bob, bob_second_device)

